packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -264,6 +262,7 @@ class ChangeNotifier implements Listenable {
       _debugDisposed = true;
       return true;
     }());
+    _listeners = const <VoidCallback?>[];
```

Comment:
Set _count to 0 as well and then the for loop in removeListener would just immediately terminate without the need for the `_debugDisposed` guard?